docs/how-to/virtual_memory.rst

```diff
@@ -0,0 +1,102 @@
+
+.. _memory_allocation_virtual_memory:
+Memory Allocation
+=================
+
+Previously, you called the ``hipMalloc`` function to allocate a chunk of memory on the device. ``hipMalloc`` returns a pointer, which is immediately usable after the call. When using virtual memory this is separated into multiple steps using the ``hipMemCreate``, ``hipMemAddressReserve``, ``hipMemMap`` and ``hipMemSetAccess`` functions. In this guide you'll learn what these functions do and how you can use them for your needs.
+
+Allocate Physical Memory
+------------------------
+
+The first step is allocate the physical memory itself with the ``hipMemCreate`` function. This function accepts the size of the buffer, an ``unsigned long long`` variable for the flags, and a ``hipMemAllocationProp`` variable. ``hipMemAllocationProp`` contains the properties of the memory to be allocated, such as where the memory is physically located and what kind of shareable handles are available. If the allocation is successful, the function returns a value of ``hipSuccess``, with ``hipMemGenericAllocationHandle_t`` representing a valid physical memory allocation. The allocated memory size has to be aligned with the granularity appropriate for the properties of the allocation. You can use the ``hipMemGetAllocationGranularity`` function to determine the correct granularity.
+
+.. code-block:: cpp
+
+    size_t granularity = 0;
+    hipMemGenericAllocationHandle_t allocHandle;
+    hipMemAllocationProp prop = {};
+    prop.type = HIP_MEM_ALLOCATION_TYPE_PINNED;
+    prop.location.type = HIP_MEM_LOCATION_TYPE_DEVICE;
+    prop.location.id = currentDev;
+    hipMemGetAllocationGranularity(&granularity, &prop, HIP_MEM_ALLOC_GRANULARITY_MINIMUM);
+    padded_size = ROUND_UP(size, granularity);
+    hipMemCreate(&allocHandle, padded_size, &prop, 0);
+
+Reserve Virtual Address Range
+-----------------------------
+
+After you have acquired the allocation of physical memory, you must map it before you can use it. To do so, you need a virtual address to map it to.  Mapping means that the physical memory allocation is available by means of the virtual address range it is mapped to. To reserve a virtual memory range, use the ``hipMemAddressReserve`` function. The size of the virtual memory must match the amount of physical memory previously allocated. You can then map the physical memory allocation to the newly-acquired virtual memory address range using the ``hipMemMap`` function.
+
+.. code-block:: cpp
+
+    hipMemAddressReserve(&ptr, padded_size, 0, 0, 0);
+    hipMemMap(ptr, padded_size, 0, allocHandle, 0);
+
+Set Memory Access
+-----------------
+
+Finally, you have to enable access to the memory. This is achieved by using the ``hipMemSetAccess`` function. It accepts the pointer to the virtual memory, the size, and a ``hipMemAccessDesc`` descriptor as parameters.
```

Comment:
```suggestion
Finally, use the ``hipMemSetAccess`` function to enable memory access. It accepts the pointer to the virtual memory, the size, and a ``hipMemAccessDesc`` descriptor as parameters.
```

Reply:
Updated.

Reply:
It looks like a couple of stray characters were added after this sentence by mistake. I'll indicate the error on the actual file as well.

Reply:
Oh, thanks! Fixed.